roborazzi-compose-preview-scanner-support/src/main/java/com/github/takahirom/roborazzi/RoborazziPreviewScannerSupport.kt

```diff
@@ -19,30 +21,59 @@ fun ComposablePreview<AndroidPreviewInfo>.captureRoboImage(
 
 @ExperimentalRoborazziApi
 interface ComposePreviewTester<T : Any> {
+  data class Options(
+    val testLifecycleOptions: TestLifecycleOptions = JUnit4TestLifecycleOptions(),
+    val scanOptions: ScanOptions = ScanOptions(emptyList()),
+  ) {
+    interface TestLifecycleOptions
+    class JUnit4TestLifecycleOptions(
+      // Used from generated tests
+      @Suppress("unused") val testRule: TestRule = object : TestWatcher() {},
```

Comment:
@yschimke 
Thanks as always. It's a little unrelated to this PR, but to address the feedback, I introduced a test rule option to support lifecycle hooks in Preview Support. Could you take a look?
https://github.com/takahirom/roborazzi/issues/446#issuecomment-2247203630

Reply:
I'd like to see a sample of using it.

Does every test get it's own instance of TestRule?

They can be stateful https://github.com/junit-team/junit4/blob/ed47b7f487bafa48cff47f051af81a004cd36049/src/main/java/org/junit/rules/ErrorCollector.java

Reply:
Thanks for your great review! I changed testRule to testRuleFactory to create a test rule for each test and added integration tests. Could you check it out?
https://github.com/takahirom/roborazzi/pull/445/commits/9e20058c9daefbdc44b9752bb974a0cb61b4860e#diff-caa31d18b2a269ce09960b3943b693c5dcdd8c57ee34cd096b4a7ce27b6423adR8

Reply:
Yeah, I think that looks good